### PR TITLE
Updates to add Nanopore sequencing data

### PR DESCRIPTION
**Pull request by Snehal1894**
This change also involved model change. So migration required in cLIMS.

1.One button can be used to upload all the file paths.
2. Exp Name, Run Name and File Path are compulsory fields and data should be present for all the entries.
3. The format has to be followed as mentioned in .csv file template to be downloaded from import page
4. Based on sequencing run platform (Nanopore), it is decided whether the file paths are for nanopore or other.
5. If the run is "Nanopore" , the Type column has be filled for all entried.
6. Missing data is allowed for md5,sha256 and backup path.
7. At a time, entries from only sequencing run should be entered.

### CODE REVIEW

File: cLIMS/organization/importSeqFiles.py [hunk 1]
```diff
@@ -11,78 +11,79 @@ def importSeqFiles(request,prj_pk):
...
+            bpath=True
+        runDict[path].append(BackupPath)
             
-            if(path in exsisting_files):
-               dupDict[path]=runDict[path]
-               del runDict[path]
-        count+=1
+        if(path in exsisting_files):
+           dupDict[path]=runDict[path]
+           del runDict[path]
```
Comment (sergpolly) [inline cLIMS/organization/importSeqFiles.py:77]: please display a message here - that some of the files user is trying to add - already exist in the database ...

File: cLIMS/organization/static/siteWide/css/bootstrap.css [hunk 2]
```diff
@@ -6754,4 +6754,14 @@ button.close {
     display: none !important;
   }
 }
+
+.overflow-wrap {
+  width: 240px;
+  border: 1px solid #000;
```
Comment (sergpolly) [inline cLIMS/organization/static/siteWide/css/bootstrap.css:6760]: remove - so there is no diff here

File: cLIMS/organization/templates/importNanopore.html [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        {% endfor %}
+{% endif %}
+
+
+ {% if dupDict %}
+<span class="tableHead">
+	Following entries already exists:
+</span>
+
+<table class="table table-bordered table-hover table-responsive">
```
Comment (sergpolly) [inline cLIMS/organization/templates/importNanopore.html:26]: this should not be here

File: cLIMS/organization/views.py [hunk 4]
```diff
@@ -1820,22 +1820,23 @@ def dispatch(self, request, *args, **kwargs):
 @class_login_required        
 class CreateSequencingFiles(View):
     def post(self,request,prj_pk):
+        print("In seq")
```
Comment (sergpolly) [inline cLIMS/organization/views.py:1823]: get rid of print statements

File: cLIMS/organization/forms.py [hunk 5]
```diff
@@ -57,9 +57,8 @@ class ExperimentForm(ModelForm):
     class Meta:
         model = Experiment
         exclude = ('project','experiment_biosample','experiment_fields','dcic_alias','update_dcic','finalize_dcic_submission',)
-        fields = ['experiment_name','biocore_name','bio_rep_no','tec_rep_no','biosample_quantity','biosample_quantity_units','protocol','type','variation','experiment_enzyme',
+        fields = ['experiment_name','biocore_name','bio_rep_no','tec_rep_no','biosample_quantity','biosample_quantity_units','collection_method','protocol','type','variation','experiment_enzyme',
```
[the remark this answers is not in the document]
Comment (Snehal1894) [reply cLIMS/organization/forms.py:60]: No.This was added to include the method of collection after the 4DN-II naming associated confusion.

File: cLIMS/organization/importSeqFiles.py [hunk 6]
```diff
@@ -11,78 +11,79 @@ def importSeqFiles(request,prj_pk):
...
     dupDict=OrderedDict()
-    orderList=[]
+    orderList=["Experiment Name","Sequencing Run Name","md5sum","sha256sum","BackupPath"]
+    nano=["Experiment Name","Sequencing Run Name","Type","md5sum","sha256sum","BackupPath"]
     context = {}
-    count=1
-    for line in lines:
+    rn=lines[1].split(',')[1]
+    platform=SequencingRun.objects.get(run_name=rn).run_sequencing_instrument.choice_name
+    print(platform)
```
[the remark this answers is not in the document]
Comment (Snehal1894) [reply cLIMS/organization/importSeqFiles.py:25]: No they wont. Was just using it for debugging. Removed then in next commit.

File: cLIMS/organization/importSeqFiles.py [hunk 7]
```diff
@@ -11,78 +11,79 @@ def importSeqFiles(request,prj_pk):
...
+    md5sum=sha256=ftype=bpath=False
+    for line in lines[1:]:
         line=line.rstrip("\r")
```
[the remark this answers is not in the document]
Comment (Snehal1894) [reply cLIMS/organization/importSeqFiles.py:28]: Yes the upload should have all the columns in the given order.

File: cLIMS/organization/importSeqFiles.py [hunk 8]
```diff
@@ -11,78 +11,79 @@ def importSeqFiles(request,prj_pk):
...
-            
-            else:
-                messages.error(request,"something is not correct with your input file.")
+            messages.error(request,"something is not correct with your input file; check Experiment Name, Run Name and FilePath")
+            runDict.clear()
+            dupDict.clear()
+            break
+        
+        if(platform=="Nanopore"):
+            print("In Nanopore condition")
```
[the remark this answers is not in the document]
Comment (Snehal1894) [reply cLIMS/organization/importSeqFiles.py:48]: Removed in next commit

File: cLIMS/organization/models.py [hunk 9]
```diff
@@ -55,6 +55,14 @@ class Experiment(References):
         ('μg', 'μg'),
         ('ml', 'ml'),
         ('cells', 'cells'),
+        ('nuclei', 'nuclei'),
+    )
+    COLLECTION_CHOICES = (
+        ('', ''),
```
[the remark this answers is not in the document]
Comment (Snehal1894) [reply cLIMS/organization/models.py:61]: Nope, just a bug that we caught accidently.

**Pull request merged**